### PR TITLE
Logs agent - retry more HTTP status codes

### PR DESCRIPTION
**Pull request by gh123man**
### What does this PR do?

Followup to this pr: https://github.com/DataDog/datadog-agent/pull/10870
We also want to retry logs on 404. 

### Describe how to test/QA your changes

UT's cover this pretty well so im not sure manual QA is worth the effort: 

- Spin up a mock server that implements the intake API endpoint and make it return 404. 
- Send some logs and watch the agent logs
- Look for the logs messages: `%s: sleeping until %v before retrying"`

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by kayayarai — APPROVED** [review]

File: pkg/logs/client/http/destination_test.go [hunk 1]
```diff
@@ -67,38 +67,15 @@ func TestDestinationSend200(t *testing.T) {
...
-			break
-		}
-	}
-	<-output
-
-	server.Stop()
+func TestRetries(t *testing.T) {
+	retryTest(t, 500)
+	retryTest(t, 429)
+	retryTest(t, 404)
```
Comment (djmitche) [inline pkg/logs/client/http/destination_test.go:73]: 👍🏽 ❤️

**Review by djmitche — APPROVED** [review]

**Comment by gh123man** [comment]
This needs a re-review since I changed the core of the PR. CC @djmitche

**Comment by djmitche** [comment]
image
 :)

File: pkg/logs/client/http/destination.go [hunk 2]
```diff
@@ -298,14 +298,17 @@ func (d *Destination) unconditionalSend(payload *message.Payload) (err error) {
...
 	}
-	if resp.StatusCode == 429 || resp.StatusCode >= 500 {
-		// the server could not serve the request, most likely because of an
-		// internal error or, (429) because it is overwhelmed
-		return client.NewRetryableError(errServer)
-	} else if resp.StatusCode >= 400 {
+	if resp.StatusCode == 400 ||
+		resp.StatusCode == 401 ||
+		resp.StatusCode == 403 ||
+		resp.StatusCode == 413 {
```
Comment (djmitche) [inline pkg/logs/client/http/destination.go:304]: It might be clearer to use http.StatusRequestEntityTooLarge, etc. here?

**Review by djmitche — APPROVED** [review]

**Pull request merged**